Create separate fake iso639 "any" entry

jstebbins opened the pull request:
Allows us to distinguish a selection of "any" which means match any
language from "und" which means the language is not known.

Fixes https://github.com/HandBrake/HandBrake/issues/731

I think I've correctly modified OSX code to handle the changes.  Someone should review and test.
Windows probably needs some additional tweaks.

**Test on:**

- [x] Windows 10+  (via MinGW)
- [x] macOS 10.13+
- [x] Ubuntu Linux

jstebbins committed:
Create separate fake iso639 "any" entry

Allows us to distinguish a selection of "any" which means match any
language from "und" which means the language is not known.

Fixes https://github.com/HandBrake/HandBrake/issues/731

galad87 commented:
The Mac changes are ok, but now every updated preset AudioLanguageList is empty, so no track is selected for built-in presets and others.

edit: it was empty before too, but we defaulted to the old "any".

jstebbins commented:
@galad87 I don't understand your comment.  What's an "updated preset AudioLanguageList" in this context?  Can you describe it from a UI perspective?  What do you expect and what is it doing?

jstebbins commented:
Oh, I think I see.  When scanning a source, no audio tracks are selected when "Selected Languages" has not been set up in "Track Selection" tab.  I'll fix that.  Is there more?

jstebbins committed:
Fix empty audio selection when "Selected Languages" is empty

galad87 commented:
No, that was it. 😅

jstebbins committed:
LinGui: fix incorrect handling of "unknown" audio lang

fix language list index lookup

jstebbins committed:
preset: fix behaviour of empty audio list failsafe

It was adding all tracks instead of just one.  Since this is a failsafe
it should just add a single track regardless of what "Selection
Behavior" says.

jstebbins committed:
preset: change last ditch audio selection behavior

Previously, it added "any" audio if the output track list was empty.
Now only adds "any" audio if AudioLanguageList is empty.

This allow e.g. "English or none" audio selections.  If the user wants
"English or any", they can specify [English, Any] in AudioLanguageList.

bradleysepos commented:
Tested to the best of my ability.

jstebbins commented:
Pushed:
e88c6d5b7 preset: change last ditch audio selection behavior
ed346b465 Create separate fake iso639 "any" entry
